packages/astro/src/core/build/plugins/plugin-ssr.ts

```diff
@@ -37,11 +42,6 @@ function vitePluginSSR(
 				inputs.add(getVirtualModulePageName(ASTRO_PAGE_MODULE_ID, pageData.component));
 			}
 
-			const adapterServerEntrypoint = options.settings.adapter?.serverEntrypoint;
-			if (adapterServerEntrypoint) {
-				inputs.add(adapterServerEntrypoint);
```

Comment:
So that the adapter will be built to a separate module and get executed as one of the first things. With the addition of using `crypto` we need the Node adapter to run early because it polyfills in Node 18. Without this the adapter will be inlined into this main module and not execute in time. Here's the scenario:

```js
import 'package-depends-on-crypto';

// Adapter code here
globalThis.crypto = ...
```

This throws. because an imported dependency executes first. With this change however:

```js
import './adapter.mjs';
import 'package-depends-on-crypto';
```

This now works, because the adapter bundle is the first thing to execute, allowing it to polyfill anything it needs to.

Reply:
Isn't it just us that uses `globalThis.crypto`? Could we ponyfill it ourselves at the point of use instead of relying on a global polyfill?

Reply:
Not easily, no, you need to import `node:crypto` which doesn't work in non-Node environments.

Reply:
Only if `globalThis.crypto` isn't defined, which it would be on those runtimes

Reply:
Not sure I follow, `globalThis.crypto` is defined in ever environment (to my knowledge) other than Node 18. How would you suggest importing it without it causing bugs when Cloudflare / Deno do their bundling?

Reply:
I meant externalising it and importing it dynamically if it's undefined. Deno is fine because it supports `node:crypto` anyway. I don't know enough about how bundling works for Cloudflare to say if it would work.

Reply:
Yeah, basically. That makes sense. I think the real fix here is to work out why rollup is unhappy with Netlify's entrypoint being external